asset/asset.go

```diff
@@ -658,6 +686,14 @@ type Asset struct {
 	// its unique ID within the Taproot Asset protocol.
 	Genesis
 
+	// Type uniquely identifies the type of Taproot asset.
+	ExplicitType Type
```

Comment:
Why's this needed? 

I would imagine that we can actually keep the `Genesis` struct here as is. The difference would be that upon decode, you would then "supplement" this information from the DB or elsewhere. In most cases, you still want to know this information, but you might obtain it in an extra step. The universe changes made a similar gap where we already know the genesis before decoding the asset, so we supplement that as needed.

Reply:
AH true, I forgot about that part of offline discussion.

I think the group key field would still change though, since the sig is being moved into the witness.

Reply:
Yeah true, the key changes as now it's _just_ a key and not also the sig that comes along side it.